server/plugin/api_test.go

```diff
@@ -266,21 +269,44 @@ func TestHandleLink(t *testing.T) {
 			err:        errors.New("mockError"),
 			statusCode: http.StatusBadRequest,
 		},
+		{
+			description: "HandleLink: project is already linked",
+			body: `{
+				"organization": "mockOrganization",
+				"project": "mockProject"
+				}`,
+			statusCode: http.StatusOK,
+			projectList: []serializers.ProjectDetails{
+				{
+					MattermostUserID: "mockMattermostUserID",
+					ProjectName:      "mockProject",
+					OrganizationName: "mockOrganizationName",
+					ProjectID:        "mockProjectID",
+				},
+			},
+			isProjectLinked: true,
+		},
 	} {
 		t.Run(testCase.description, func(t *testing.T) {
 			mockAPI.On("LogError", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"))
 			mockAPI.On("GetDirectChannel", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(&model.Channel{}, nil)
 			mockAPI.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(&model.Post{}, nil)
 
+			monkey.PatchInstanceMethod(reflect.TypeOf(p), "IsProjectLinked", func(*Plugin, []serializers.ProjectDetails, serializers.ProjectDetails) (*serializers.ProjectDetails, bool) {
+				return &serializers.ProjectDetails{}, testCase.isProjectLinked
+			})
+
 			if testCase.statusCode == http.StatusOK {
-				mockedClient.EXPECT().Link(gomock.Any(), gomock.Any()).Return(&serializers.Project{}, testCase.statusCode, testCase.err)
-				mockedClient.EXPECT().CheckIfUserIsProjectAdmin(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(http.StatusOK, nil)
 				mockedStore.EXPECT().GetAllProjects("mockMattermostUserID").Return(testCase.projectList, nil)
-				mockedStore.EXPECT().StoreProject(&serializers.ProjectDetails{
-					MattermostUserID: "mockMattermostUserID",
-					ProjectName:      "Mockproject",
-					OrganizationName: "mockorganization",
-				}).Return(nil)
+				if !testCase.isProjectLinked {
+					mockedClient.EXPECT().Link(gomock.Any(), gomock.Any()).Return(&serializers.Project{}, testCase.statusCode, testCase.err)
+					mockedClient.EXPECT().CheckIfUserIsProjectAdmin(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(http.StatusOK, nil)
+					mockedStore.EXPECT().StoreProject(&serializers.ProjectDetails{
+						MattermostUserID: "mockMattermostUserID",
```

Comment:
you can create a test util function here we will return you mock projectDetails and use that everywhere